swayfmt/src/items/item_impl/mod.rs

```diff
@@ -38,14 +38,27 @@ impl Format for ItemImpl {
             where_clause.format(formatted_code, formatter)?;
             formatter.shape.code_line.update_where_clause(true);
         }
-        Self::open_curly_brace(formatted_code, formatter)?;
+
         let contents = self.contents.get();
-        for item in contents.iter() {
-            write!(formatted_code, "{}", formatter.indent_str()?,)?;
-            item.format(formatted_code, formatter)?;
-            writeln!(formatted_code)?;
+        if contents.is_empty() {
+            Self::open_curly_brace(formatted_code, formatter)?;
+            let comments = write_comments(formatted_code, self.span().into(), formatter)?;
+            if !comments {
+                formatter.unindent();
+                if formatted_code.ends_with('\n') {
+                    formatted_code.pop();
+                }
```

Comment:
Please do not use `pop()`, or other destructive methods if possible, especially on the returned formatted string.

Reply:
My approach was tu use `truncate` but I need to make the current string smaller. I'll check #5149 to check how you did it.

Reply:
@eureka-cpu I removed the destructive method in a8b8e0379